src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -1719,7 +1719,7 @@ MethodHandle setVarargs(MemberName member) throws IllegalAccessException {
         try {
             return this.withVarargs(true);
         } catch (IllegalArgumentException ex) {
-            throw member.makeAccessException("cannot make variable arity", null);
+            throw new IllegalAccessException("cannot make variable arity: " + this + " because " + ex.getMessage());
```

Comment:
This now drops the member information (owner and name) from the message. Is that intended?

In addition, the original message appears to include the method type, which can be used to diagnose why varargs is not possible (not an array for trailing parameter). I fail to see what extra information this patch offers that actually makes debugging easier as claimed in the JBS issue.

Reply:
Good catch.  It's not intended.  Fixed.

I think in general it would be useful to make it clear in the exception message of the reason especially for those who are new in using method handles.  Looking it again, it can simply hardcode the message (more friendly than IAE exception message):

```
java.lang.IllegalAccessException: cannot make variable arity: MyClass.m(Object[],int)void/invokeStatic because not an array type: int

vs

java.lang.IllegalAccessException: cannot make variable arity: trailing parameter type is not an array type: MyClass.m(Object[],int)void/invokeStatic
```